wrp/header_wrp.go

```diff
@@ -66,16 +72,16 @@ func HeaderToWRP(header http.Header) (*Message, error) {
 	} else if msg.Type == SimpleRequestResponseMessageType || msg.Type == SimpleEventMessageType ||
 		msg.Type == CreateMessageType || msg.Type == RetrieveMessageType || msg.Type == UpdateMessageType ||
 		msg.Type == DeleteMessageType {
-		return nil, fmt.Errorf("%s", "Invalid Source header string")
+		return nil, fmt.Errorf("%s", ErrInvalidSource)
 	}
 
 	// TransactionUuid is mandatory for SimpleRequestResponse and CRUD
-	if transUuid := header.Get(TrasactionUuidHeader); !strings.EqualFold(transUuid, "") {
+	if transUuid := header.Get(TransactionUuidHeader); !strings.EqualFold(transUuid, "") {
```

Comment:
Remove all validation from this code.  This block of code should only worry about translating headers into messages.  Let the `wrp` package handle validation (which already exists).